### PR TITLE
Create EnclaveInstance using validators

### PR DESCRIPTION
**Pull request by snwagh**
Comments: 
* We are not setting the `SyftObject` `id` (base class) right now (commented out) and thus `node_uid` and `id` are different, let me know if I should change the behaviour.

Tested
* Ran `00-do-setup-domain.ipynb`

Closes [1510](https://github.com/OpenMined/Heartbeat/issues/1510)

### CODE REVIEW

File: packages/syft/src/syft/service/enclave/enclave.py [hunk 1]
```diff
@@ -45,20 +45,25 @@ class EnclaveInstance(SyftObject):
...
+    def initialize_values(cls, values: dict[str, Any]) -> dict[str, Any]:
+        if "route" in values:
+            route = values["route"]
+            metadata = login_as_guest(url=route.host_or_ip, port=route.port).metadata
+            if not metadata:
+                raise SyftException("Failed to fetch metadata from the node")
+
+            values.update(
+                {
+                    # 'id': UID(metadata.id),
```
Comment (itstauq) [inline packages/syft/src/syft/service/enclave/enclave.py]: `id` is the identifier for this object's instance in the owner domain, while `node_uid` identifies the Enclave node that this object points to.

Comment (itstauq) [reply packages/syft/src/syft/service/enclave/enclave.py]: Since this comment implies the wrong thing, let's remove it to avoid any confusion.

**Review by itstauq — APPROVED** [review]
I've added a small comment. Overall, this PR looks great to me. Approved! 💪🏻

**Comment by rasswanth-s** [comment]
Well done @snwagh

**Pull request merged**